packages/strapi-plugin-content-manager/admin/src/containers/ListView/reducer.js

```diff
@@ -88,10 +88,18 @@ const listViewReducer = (state = initialState, action) =>
 
         if (!value) {
           const { metadatas, attributes } = state.contentType;
+          let metas = metadatas[name].list;
+
+          if (attributes[name].type === 'relation') {
+            const mainField = metadatas[name].edit.mainField;
+            const { relationType } = attributes[name];
```

Comment:
This is not needed you have the relationType information in the fieldSchema object.